addons/sale_timesheet/controllers/portal.py

```diff
@@ -59,6 +59,11 @@ def portal_my_tasks_invoices(self, task_id=None, page=1, date_begin=None, date_e
 
 class SaleTimesheetCustomerPortal(TimesheetCustomerPortal):
 
+    def _prepare_portal_layout_values(self):
+        values = super()._prepare_portal_layout_values()
+        values['access_sale_order_ids'] = request.env['sale.order'].search([]).ids
```

Comment:
Since you just need  the ids, you could call the `_search` method instead. 🙂 
```suggestion
        values['access_sale_order_ids'] = list(request.env['sale.order']._search([]))
```